pkg/login/social/generic_oauth.go

```diff
@@ -244,13 +271,13 @@ func (s *SocialGenericOAuth) extractToken(data *UserInfoJson, token *oauth2.Toke
 		return false
 	}
 
-	err = json.Unmarshal(payload, data)
+	err = json.Unmarshal(payload, &data)
 	if err != nil {
 		s.log.Error("Error decoding id_token JSON", "payload", string(payload), "err", err)
 		return false
 	}
 
-	email := s.extractEmail(data)
+	email := s.extractEmail(&UserInfoJson{}, payload)
```

Comment:
Should this be:
```suggestion
	email := s.extractEmail(&data, payload)
```

Reply:
Yes, thank you. It seems I may have forgot some of the context of since picking up this PR, but I can sanity check everything once over again before we merge by early next week.